wdl-analysis/src/types/v1.rs

```diff
@@ -1643,11 +1689,11 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
         severity: Severity,
     ) {
         let (label, span, fix) = match target.text() {
-            "select_first" => {
-                let ty = &arguments[0]
-                    .as_array()
-                    .expect("type should be an array")
-                    .element_type;
+            "select_first" if !arguments.is_empty() => {
```

Comment:
`check_unnecessary_call` is only ever called upon a successful binding of a function, which means it can make assumptions about the number of arguments and their types here.

Like above, I'd rather panic if the invariant is broken rather than silently not emit a diagnostic.

Therefore, I think we can revert the changes to this function.